### PR TITLE
Fix bed corruption

### PR DESCRIPTION
**Pull request by ishitatsuyuki**

### CODE REVIEW

File: src/pocketmine/block/Bed.php [hunk 1]
```diff
@@ -125,30 +125,26 @@ public function place(Item $item, Block $block, Block $target, $face, $fx, $fy,
...
-			if($blockNorth->getId() === $this->id and $blockNorth->meta !== 0x08){ //Checks if the block ID and meta are right
-				$this->getLevel()->setBlock($blockNorth, new Air(), true, true);
-			}elseif($blockSouth->getId() === $this->id and $blockSouth->meta !== 0x08){
-				$this->getLevel()->setBlock($blockSouth, new Air(), true, true);
-			}elseif($blockEast->getId() === $this->id and $blockEast->meta !== 0x08){
-				$this->getLevel()->setBlock($blockEast, new Air(), true, true);
-			}elseif($blockWest->getId() === $this->id and $blockWest->meta !== 0x08){
-				$this->getLevel()->setBlock($blockWest, new Air(), true, true);
+			$next = $this->getSide($sides[$this->meta]);
+			if($next->getId() === $this->id and ($next->meta | 0x08) == $this->meta){ //Checks if the block ID and meta are right
```
Comment (ishitatsuyuki) [inline src/pocketmine/block/Bed.php]: @dog194 Use === here.

**Comment by ishitatsuyuki** [comment]
Rejected: please resolve the comments.

**Comment by ishitatsuyuki** [comment]
Looks good to me. This should work.

**Comment by dog194** [comment]
+1

**Pull request merged**